authorization/http.go

```diff
@@ -36,14 +36,13 @@ func WithAuthorizers(authorizers map[string]rbac.Authorizer, permission rbac.Per
 			if !ok {
 				groups = []string{}
 			}
-			a, ok := authorizers[tenant]
-			if !ok {
+			if authorizer == nil {
```

Comment:
Again, the check and the error message do not align here. I don't think we need to check that the function pointer is not nil. If we did, this world certainly be a programmer error rather than a runtime error and thus we should panic or return an error to the caller rather than an HTTP error to the user. Note that the reason for this check in the first place is that the user could supply a nonexistent tenant name, which would result in a failed lookup. You can't have a failed lookup with your new model.

Reply:
Makes sense. Moved the check for 'nil' auth. closer to its creation ( newTenant() / main)  and fail the tenant registration if auth turns out to be 'nil'